packages/clerk-js/src/core/fapiClient.ts

```diff
@@ -184,16 +184,26 @@ export default function createFapiClient(clerkInstance: Omit<Clerk, 'proxyUrl'>)
     const overwrittenRequestMethod = method === 'GET' ? 'GET' : 'POST';
     let response: Response;
     const urlStr = requestInit.url.toString();
+    const fetchOpts: FapiRequestInit = {
+      ...requestInit,
+      credentials: 'include',
+      method: overwrittenRequestMethod,
+    };
 
     try {
-      response = beforeRequestCallbacksResult
-        ? await fetch(urlStr, {
-            ...requestInit,
-            credentials: 'include',
-            method: overwrittenRequestMethod,
-          })
-        : // Mock an empty json response
-          new Response('{}', requestInit);
+      if (beforeRequestCallbacksResult) {
+        response =
+          // retry only on GET requests for safety
+          overwrittenRequestMethod === 'GET'
+            ? await runWithExponentialBackOff(() => fetch(urlStr, fetchOpts), {
+                maxRetries: 6,
+                firstDelay: 500,
+                maxDelay: 5000,
```

Comment:
ℹ️  This config results in the following timeline:

**0sec -- 0,5sec -- 1,5sec -- 3,5sec -- 7,5sec -- 15,5sec -- THROW.**